ui/admin/app/components/form/field/list-wrapper/key-value/index.hbs

```diff
@@ -3,39 +3,76 @@
   SPDX-License-Identifier: MPL-2.0
 }}
 {{#let (unique-id) (unique-id) as |keyHeaderID valueHeaderID|}}
-
   <Hds::Table class='list-wrapper-field' name={{@name}}>
     <:head as |H|>
       <H.Tr>
-        <H.Th id={{keyHeaderID}}>{{t 'form.key.label'}}</H.Th>
-        <H.Th id={{valueHeaderID}}>{{t 'form.value.label'}}</H.Th>
+        <H.Th id={{keyHeaderID}}>
+          {{#if @keyLabel}}
+            {{@keyLabel}}
+          {{else}}
+            {{t 'form.key.label'}}
+          {{/if}}</H.Th>
+        <H.Th id={{valueHeaderID}}>
+          {{#if @valueLabel}}
+            {{@valueLabel}}
+          {{else}}
+            {{t 'form.value.label'}}
+          {{/if}}
+        </H.Th>
         <H.Th>
           {{t 'titles.actions'}}
-
         </H.Th>
       </H.Tr>
     </:head>
     <:body as |B|>
       {{#each this.options as |option index|}}
         <B.Tr>
           <B.Td>
-            <Hds::Form::TextInput::Field
-              @value={{option.key}}
-              @type='text'
-              disabled={{@disabled}}
-              aria-labelledby={{keyHeaderID}}
-              {{on 'input' (set-from-event option 'key')}}
-            />
+            {{yield
+              (hash
+                text=(component
+                  'form/field/list-wrapper/key-value/text'
+                  disabled=@disabled
+                  value=option.key
+                  ariaLabel=keyHeaderID
+                  setContext=(set-from-event option 'key')
+                )
+                select=(component
+                  'form/field/list-wrapper/key-value/select'
+                  disabled=@disabled
+                  value=option.key
+                  selectOptions=@selectOptions
```

Comment:
Shouldn't `selectOptions` be passed directly to the component rather than from the key value component? If I'm reading this right we're passing in `selectOptions` from KeyValue which would mean we can't use two selects?

Reply:
Good point! With the current design, it is not possible to have two dropdowns. Passing directly to the component does not get yielded(maybe I'm doing something wrong here?). What do you think about adding separate key and value select options to the keyValue instead of having a single `selectoptions`?

Reply:
Hmm I was able to get it to yield it but I had to remove `showNewRow` and `selectOptions` since they were using `selectOptions`, is that what you were seeing?

Reply:
yup, it works without those, but I can't access the consumer's nested component args in the `keyvalue` component. This means we won't be able to have unique rows which are necessary for the credential library.

Reply:
I was thinking we could let the consumer handle that logic instead of baking it into the component, what do you think? That means we'd have to pass in the necessary logic for that one consumer

Reply:
The issue is we have two separate new and existing row blocks in the component and a single `selectOptions` arg from the consumer. If we were to handle this logic(filter out duplicates) in the consumer, the consumer wouldn't know if they are modifying the new or the existing row.

Reply:
moved to consumer as discussed